packages/server/src/ApolloServer.ts

```diff
@@ -214,6 +220,19 @@ export class ApolloServer<in out TContext extends BaseContext = BaseContext> {
 
     const isDev = nodeEnv !== 'production';
 
+    if (
+      config.cache &&
+      config.cache !== 'bounded' &&
```

Comment:
Can we log / warn when this is set? It'd be nice to re-remove this for good in v5

Reply:
You think so? I feel like it's not a huge deal and not worth bugging folks about (and if we have a very short migration guide for v5 then it's fine to remove it then without an explicit deprecation period).

Reply:
Yeah I don't feel too strongly about it, I guess the nice thing about the warning is it's simple and actionable: "you have a line of code that you can delete". I'm ok with leaving it though.